chain-api/src/types/dexDtos.ts

```diff
@@ -0,0 +1,576 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import BigNumber from "bignumber.js";
+import { IsNotEmpty, IsNumber, IsOptional, IsString, Min } from "class-validator";
+
+import { BigNumberProperty } from "../validators";
+import { TokenBalance } from "./TokenBalance";
+import { TokenClassKey } from "./TokenClass";
+import { PositionInPool } from "./dexTypes";
+import { ChainCallDTO } from "./dtos";
+
+export class CreatePoolDto extends ChainCallDTO {
+  @IsNotEmpty()
+  public token0: TokenClassKey;
+  @IsNotEmpty()
+  public token1: TokenClassKey;
+  @IsNotEmpty()
+  public fee: number;
+  @BigNumberProperty()
+  @IsNotEmpty()
+  public initialSqrtPrice: BigNumber;
+
+  @IsNumber()
+  protocolFee: number;
+
+  constructor(
+    token0: TokenClassKey,
+    token1: TokenClassKey,
+    fee: number,
+    initialSqrtPrice: BigNumber,
+    protocolFee: number = 0
+  ) {
+    super();
+    this.token0 = token0;
+    this.token1 = token1;
+    this.fee = fee;
+    this.initialSqrtPrice = initialSqrtPrice;
+    this.protocolFee = protocolFee;
+  }
+}
+
+export class PositionDto extends ChainCallDTO {
+  @IsNotEmpty()
+  public token0: TokenClassKey;
+  @IsNotEmpty()
+  public token1: TokenClassKey;
+  @IsNotEmpty()
+  public fee: number;
+  @IsNotEmpty()
+  public owner: string;
+  @IsNotEmpty()
+  public tickLower: number;
+  @IsNotEmpty()
+  public tickUpper: number;
+  @IsNotEmpty()
+  @BigNumberProperty()
+  public liquidity: BigNumber;
+
+  constructor(
+    token0: TokenClassKey,
+    token1: TokenClassKey,
+    fee: number,
+    owner: string,
+    tickLower: number,
+    tickUpper: number,
+    liquidity: BigNumber
+  ) {
+    super();
+    this.token0 = token0;
+    this.token1 = token1;
+    this.fee = fee;
+    this.owner = owner;
+    this.tickLower = tickLower;
+    this.tickUpper = tickUpper;
+    this.liquidity = liquidity;
+  }
+}
+
+export class QuoteExactAmountDto extends ChainCallDTO {
+  @IsNotEmpty()
+  public token0: TokenClassKey;
+  @IsNotEmpty()
+  public token1: TokenClassKey;
+  @IsNotEmpty()
+  public fee: number;
+  @IsNotEmpty()
+  public zeroForOne: boolean;
+  @BigNumberProperty()
+  @IsNotEmpty()
+  public amount: BigNumber;
+
+  constructor(
+    token0: TokenClassKey,
+    token1: TokenClassKey,
+    fee: number,
+    amount: BigNumber,
+    zeroForOne: boolean
+  ) {
+    super();
+    this.token0 = token0;
+    this.token1 = token1;
+    this.fee = fee;
+    this.amount = amount;
+    this.zeroForOne = zeroForOne;
+  }
+}
+
+export class SwapDto extends ChainCallDTO {
+  @IsNotEmpty()
+  public token0: TokenClassKey;
+  @IsNotEmpty()
+  public token1: TokenClassKey;
+  @IsNotEmpty()
+  public fee: number;
+  public recipient: string;
```

Comment:
```suggestion
  @IsNotEmpty()
  public recipient: string;
```
This needs a decorator, probably `@IsNotEmpty`